### PR TITLE
Rabbitmq tests

### PR DESCRIPTION
**Pull request by maheeka**
rabbitmq integration tests

### CODE REVIEW

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/clients/rabbitmqclient/RabbitMQConsumerClient.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        try {
+            //Throws an exception if the channel is not available.
+            channel.exchangeDeclarePassive(exchangeName);
+        } catch (IOException e) {
+            //Channel is closed if exchange is not available. Therefore, create the channel here again.
+            if (!channel.isOpen()) {
+                channel = connection.createChannel();
+            }
+            channel.exchangeDeclare(exchangeName, "direct");
```
Comment (malakasilva) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/clients/rabbitmqclient/RabbitMQConsumerClient.java]: Don't we have to catch exception here as well?

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/clients/rabbitmqclient/RabbitMQConsumerClient.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        List<String> messages = new ArrayList<String>();
+        GetResponse response = null;
+
+        while ((response = channel.basicGet(routeKey, true))!=null) {
+            messages.add(new String(response.getBody()));
+        }
+        return messages;
+    }
+
+    public void disconnect() throws IOException {
```
Comment (sandamal) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/clients/rabbitmqclient/RabbitMQConsumerClient.java]: If the first one throws and IOException, the second one will not close, so better catch the exception inside.

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/clients/rabbitmqclient/RabbitMQProducerClient.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        channel.queueBind(routeKey, exchangeName, routeKey);
+    }
+
+    public void sendBasicMessage(String message) throws IOException {
+        channel.basicPublish(exchangeName, routeKey,
+                new AMQP.BasicProperties.Builder().contentType("text/plain").build(), message.getBytes());
+    }
+
+    public void disconnect() throws IOException {
+        channel.close();
```
Comment (sandamal) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/clients/rabbitmqclient/RabbitMQProducerClient.java]: same as above

File: modules/integration/tests-integration/tests-sample/src/test/java/org/wso2/carbon/esb/samples/test/mediation/Sample18TestCase.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        } finally {
+            if (output != null) {
+                try {
+                    output.close();
+                } catch (IOException logOrIgnore) {
+                    log.error("Error while closing the connection");
+                }
+            }
+        }
+        InputStream response = connection.getInputStream();
```
Comment (sandamal) [inline modules/integration/tests-integration/tests-sample/src/test/java/org/wso2/carbon/esb/samples/test/mediation/Sample18TestCase.java]: is this stream closed?

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForEachPropertyMediatorTestCase.java [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+        OutputStream output = null;
+        try {
+            output = connection.getOutputStream();
+            output.write(query.getBytes(charset));
+        } finally {
+            if (output != null) {
+                output.close();
+            }
+        }
+        InputStream response = connection.getInputStream();
```
Comment (sandamal) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForEachPropertyMediatorTestCase.java]: stream is not closed

**Comment by maheeka** [comment]
fixed review commits and resent-pr https://github.com/wso2/product-esb/pull/173

**Pull request closed without merging**